tools/test_api.py

```diff
@@ -2272,7 +2272,17 @@ def build_tests(tests, base_source_paths, build_path, target, toolchain_name,
 
 def test_spec_from_test_builds(test_builds):
     for build in test_builds:
-        if Target.get_target(test_builds[build]['platform']).is_PSA_non_secure_target:
+        if test_builds[build]['platform'][:12] == 'NU_PFM_M2351':
+            # Translate target name 'NU_PFM_M2351'+SUFFIX to platform name
+            # 'NUMAKER_PFM_M2351' for the NuMaker-PFM-M2351 board:
+            #
+            # 1. Online database has limit of max 20 chars on target name,
+            #    so we cannot follow the naming rule 'NUMAKER_PFM_M2351'+
+            #    SUFFIX for this board.
+            # 2. The registered platform name 'NUMAKER_PFM_M2351' in mbed-os-tools
+            #    keeps unchanged for compatibility.
+            test_builds[build]['platform'] = 'NUMAKER_PFM_M2351'
```

Comment:
I'm sorry, but we cannot accept this as it changes the ID. This is likely to have serious implications throughout the build system (apart from making it very difficult to debug).

Reply:
@mark-edgeworth OK, and I still need to update platform database in mbed-os-tools to change platform name to **NU_PFM_M2351** from **NUMAKER_PFM_M2351**. Note there're multiple target names for the same board due to combinations of PSA/non-PSA and secure/non-secure as the pattern **NU_PFM_M2351_[P, NP]_[S, NS]**, stripping suffix of these target names to **NU_PFM_M2351** is still inevitable.

Reply:
Looks ok now; names fit in 20 characters.